work_queue/src/work_queue_server.c

```diff
@@ -272,12 +270,26 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
+    // what is the port chosen for the queue?
+    port = work_queue_port(queue);
+
 	struct link *server_link = link_serve(server_port);
 	if(!server_link) {
 		printf("could not serve on port %d: %s\n", server_port,strerror(errno));
 		return 1;
 	}
 
+    // what is the port chosen for the server?
+    char addr[LINK_ADDRESS_MAX];
+    int local = link_address_local(server_link, &addr, &server_port);
```

Comment:
The name of an array is the address of its first element.
So just pass `addr` not `&addr`